From ca4b7aaff690095971de96370317fdf1bcd3af8f Mon Sep 17 00:00:00 2001
From: Kalev Lember <klember@redhat.com>
Date: Fri, 26 Mar 2021 14:48:51 +0100
Subject: [PATCH] Avoid changing soname

Bumping the major version to 40 in commit
288be740b5bbb59757db5038ab435ceb7a334dcf inadvertently changed the
soname for shared libevdocument and libevview.

Fix this by making the soname calculation independent from the project
version as the library ABI hasn't actually changed.
---
 libdocument/meson.build | 2 +-
 libview/meson.build     | 2 +-
 meson.build             | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/libdocument/meson.build b/libdocument/meson.build
index 74fe2fb1..e09de7fc 100644
--- a/libdocument/meson.build
+++ b/libdocument/meson.build
@@ -138,7 +138,7 @@ cflags = [
 ]
 
 libevdocument = shared_library(
-  'evdocument' + ev_major_version.to_string(),
+  'evdocument3',
   version: ev_document_version,
   sources: sources + enum_sources + [version_header],
   include_directories: top_inc,
diff --git a/libview/meson.build b/libview/meson.build
index 20179aff..74c86d51 100644
--- a/libview/meson.build
+++ b/libview/meson.build
@@ -82,7 +82,7 @@ if enable_multimedia
 endif
 
 libevview = shared_library(
-  'evview' + ev_major_version.to_string(),
+  'evview3',
   version: ev_view_version,
   sources: sources + marshal_sources + enum_sources,
   include_directories: top_inc,
diff --git a/meson.build b/meson.build
index 1ed37e96..cafb9023 100644
--- a/meson.build
+++ b/meson.build
@@ -43,13 +43,13 @@ ev_document_version = '@0@.@1@.@2@'.format(ev_document_current, ev_document_revi
 ev_document_current_minus_age = ev_document_current - ev_document_age
 
 # Libtool version of the view library
-ev_view_current = ev_major_version
+ev_view_current = 3
 ev_view_revision = 0
 ev_view_age = 0
 ev_view_version = '@0@.@1@.@2@'.format(ev_view_current, ev_view_revision, ev_view_age)
 ev_view_current_minus_age = ev_view_current - ev_view_age
 
-ev_api_version = '@0@.0'.format(ev_major_version)
+ev_api_version = '3.0'
 
 ev_include_subdir = join_paths(ev_name, ev_api_version)
 
-- 
GitLab

